addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -408,15 +408,64 @@ def _import_fattura_pa(self, tree, invoice):
 
             # Type must be present in the context to get the right behavior of the _default_journal method (account.move).
             # journal_id must be present in the context to get the right behavior of the _default_account method (account.move.line).
-            elements = tree.xpath('//CessionarioCommittente//IdCodice')
-            company = elements and self.env['res.company'].search([('vat', 'ilike', elements[0].text)], limit=1)
-            if not company:
-                elements = tree.xpath('//CessionarioCommittente//CodiceFiscale')
-                company = elements and self.env['res.company'].search([('l10n_it_codice_fiscale', 'ilike', elements[0].text)], limit=1)
+
+            buyer_seller_info = {
+                True: {
+                    'section': 'buyer',
+                    'vat_elements': tree.xpath('//CessionarioCommittente//IdCodice'),
```

Comment:
This xpath is needlessly executed if you clicked on the `import` button on the `invoices` screen.
Just place here the `xpath` selector, and delay execution of `tree.xpath` when you do `for incoming in incoming_possibilities:` and you already know what to search.